pygeoapi/util.py

```diff
@@ -533,3 +566,155 @@ def get_envelope(coords_list: List[List[float]]) -> list:
+        )
+        LOGGER.error(msg)
+        raise CRSError(msg)
+    else:
+        return crs
+
+
+# Type for shapely geometrical objects.
+GeomObject = Union[
+    GeometryCollection,
+    LinearRing,
+    LineString,
+    MultiLineString,
+    MultiPoint,
+    MultiPolygon,
+    Point,
+    Polygon,
+]
+
+
+def get_transform_from_crs(
+    crs_in: pyproj.CRS, crs_out: pyproj.CRS,
+) -> Callable[[GeomObject], GeomObject]:
+    """ Get transformation function from two `pyproj.CRS` instances.
+
+    Get function to transform the coordinates of a Shapely geometrical object
+    from one coordinate reference system to another.
+
+    :param crs_in: Coordinate Reference System of the input geometrical object.
+    :type crs_in: `pyproj.CRS`
+    :param crs_out: Coordinate Reference System of the output geometrical
+        object.
+    :type crs_out: `pyproj.CRS`
+
+    :returns: Function to transform the coordinates of a `GeomObject`.
+    :rtype: `callable`
+    """
+    crs_transform = pyproj.Transformer.from_crs(
+        crs_in, crs_out, always_xy=True,
+    ).transform
+    return partial(shapely.ops.transform, crs_transform)
+
+
+def crs_transform(func):
+    """Decorator that transform the geometry's/geometries' coordinates of a
+    Feature/FeatureCollection.
+
+    This function can be used to decorate another function which returns either
+    a Feature or a FeatureCollection (GeoJSON-like `dict`). For a
+    FeatureCollection, the Features are stored in a ´list´ available at the
+    'features' key of the returned `dict`. For each Feature, the geometry is
+    available at the 'geometry' key. The decorated function may take a
+    'crs_transform_wkt' parameter, which accepts a `CrsTransformWkt` instance
+    as value. If the `CrsTransformWkt` instance represents a coordinates
+    transformation between two different CRSs, the coordinates of the
+    Feature's/FeatureCollection's geometry/geometries will be transformed
+    before returning the Feature/FeatureCollection. If the 'crs_transform_wkt'
+    parameter is not given, passed `None` or passed a `CrsTransformWkt`
+    instance which does not represent a coordinates transformation, the
+    Feature/FeatureCollection is returned unchanged. This decorator can for
+    example be use to help supporting coordinates transformation of
+    Feature/FeatureCollection `dict` objects returned by the `get` and `query`
+    methods of (new or with no native support for transformations) providers of
+    type 'feature'.
+
+    :param func: Function to decorate.
+    :type func: `callable`
+
+    :returns: Decorated function.
+    :rtype: `callable`
+    """
+    @functools.wraps(func)
+    def get_geojsonf(*args, **kwargs):
+        crs_transform_wkt = kwargs.get('crs_transform_wkt')
+        result = func(*args, **kwargs)
+        if crs_transform_wkt is None:
+            return result
+        else:
```

Comment:
matter of taste: I usually don't use `else` as the function only proceeds when `crs_transform_wkt ` `is not None`. Saves indentation as well.